assets/src/edit-story/components/form/dateTime/dateTime.js

```diff
@@ -52,6 +55,21 @@ function DateTime({
     hours: is12Hour ? initialHours % 12 || 12 : initialHours,
     date: selectedTime,
   });
+
+  const previousFocus = useRef(document.activeElement);
+  const handleClose = useCallback(
+    (evt) => {
+      if (previousFocus.current) {
+        previousFocus.current.focus();
+      }
+      onClose(evt);
+    },
+    [onClose]
+  );
+
+  useKeyDownEffect(forwardedRef, 'esc', handleClose);
+  useRovingTabIndex({ ref: forwardedRef });
```

Comment:
Yes, this seems to just work! 🎉

Reply:
We need all the buttons in the calendar-display to have `tabIndex=-1` from the start except the first one. Otherwise, when you start combining arrows and tabbing, you'll get some very weird results.